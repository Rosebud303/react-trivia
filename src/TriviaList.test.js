import React from 'react';
import TriviaList from './TriviaList';
import { shallow } from 'enzyme';


const mockQuestions = [{category:"Entertainment: Video Games",type:"multiple",difficulty:"hard",title:"When was the first Call of Duty title released?",correct_answer:"October 29, 2003",incorrect_answers:["December 1, 2003","November 14, 2002","July 18, 2004"],id:1},{category:"Entertainment: Books",type:"multiple",difficulty:"medium",title:"How many books are in the Chronicles of Narnia series?",correct_answer:"7",incorrect_answers:["6","8","5"],id:2},{category:"Entertainment: Film",type:"boolean",difficulty:"easy",title:"Han Solo's co-pilot and best friend, 'Chewbacca', is an Ewok.",correct_answer:"False",incorrect_answers:["True"],id:3}];
const mockQuestionCount = 2;

describe('TriviaList', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <TriviaList
        triviaQuestions={mockQuestions}
        questionCount={mockQuestionCount}
      />
    );
  });

  it('should match the snapshot with all data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});



