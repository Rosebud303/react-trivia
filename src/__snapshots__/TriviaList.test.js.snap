// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaList should match the snapshot with all data passed in 1`] = `
<ul>
  <Question
    key="1"
    question={
      Object {
        "category": "Entertainment: Video Games",
        "correct_answer": "October 29, 2003",
        "difficulty": "hard",
        "id": 1,
        "incorrect_answers": Array [
          "December 1, 2003",
          "November 14, 2002",
          "July 18, 2004",
        ],
        "title": "When was the first Call of Duty title released?",
        "type": "multiple",
      }
    }
  />
  <Question
    key="2"
    question={
      Object {
        "category": "Entertainment: Books",
        "correct_answer": "7",
        "difficulty": "medium",
        "id": 2,
        "incorrect_answers": Array [
          "6",
          "8",
          "5",
        ],
        "title": "How many books are in the Chronicles of Narnia series?",
        "type": "multiple",
      }
    }
  />
</ul>
`;

exports[`TriviaList should match the snapshot with all data passed in correctly 1`] = `
<ul>
  <Question
    key="1"
    question={
      Object {
        "category": "Entertainment: Video Games",
        "correct_answer": "October 29, 2003",
        "difficulty": "hard",
        "id": 1,
        "incorrect_answers": Array [
          "December 1, 2003",
          "November 14, 2002",
          "July 18, 2004",
        ],
        "title": "When was the first Call of Duty title released?",
        "type": "multiple",
      }
    }
  />
  <Question
    key="2"
    question={
      Object {
        "category": "Entertainment: Books",
        "correct_answer": "7",
        "difficulty": "medium",
        "id": 2,
        "incorrect_answers": Array [
          "6",
          "8",
          "5",
        ],
        "title": "How many books are in the Chronicles of Narnia series?",
        "type": "multiple",
      }
    }
  />
</ul>
`;
